### PR TITLE
added phpstan types to `CompletePackageInterface`

### PR DESCRIPTION
**Pull request by clxmstaab**

### CODE REVIEW

File: src/Composer/Package/CompletePackageInterface.php [hunk 1]
```diff
@@ -69,13 +73,15 @@ public function getHomepage();
      * Each item can contain name/homepage/email keys
      *
      * @return array
+     * @psalm-return array<array{?name: string, ?homepage: string, ?email: string}>
```
Comment (stof) [inline src/Composer/Package/CompletePackageInterface.php:76]: There is also an optional `role`

File: src/Composer/Package/CompletePackageInterface.php [hunk 2]
```diff
@@ -32,20 +33,23 @@ public function getScripts();
      * {"<type>": {<config key/values>}}
      *
      * @return array Repositories
+     * @psalm-return array<string, array{type: string, url: string, allow_ssl_downgrade: bool}>
```
Comment (stof) [inline src/Composer/Package/CompletePackageInterface.php:36]: Not all repository types have a `url`. That's more complex than that. Each type has its own fields (see the json schema)

Comment (clxmstaab) [reply src/Composer/Package/CompletePackageInterface.php]: thx for the hint. changed the url to be optional.

tbh: I dont think whether its usefull to repeat the complex types which are defined in the schema in the docblock

**Comment by clxmstaab** [comment]
@Seldaek wdyt about upping the phpstan level of composer to e.g. 3 or 4 so we can see more meaningfull errors and commit a [phpstan baseline](https://phpstan.org/blog/phpstans-baseline-feature-lets-you-hold-new-code-to-a-higher-standard) to get a green build?

File: src/Composer/Package/CompletePackageInterface.php [hunk 3]
```diff
@@ -85,6 +91,7 @@ public function getSupport();
      * Each item will contain type and url keys
      *
      * @return array
```
Comment (Seldaek) [inline src/Composer/Package/CompletePackageInterface.php:93]: Could be made `array[]` too I suppose.. it's not much info but it's something :)

File: src/Composer/Package/CompletePackageInterface.php [hunk 4]
```diff
@@ -32,20 +33,23 @@ public function getScripts();
...
      * @return array The package licenses
+     * @psalm-return string[]
      */
     public function getLicense();
 
     /**
      * Returns an array of keywords relating to the package
      *
      * @return array
+     * @psalm-return string[]
```
Comment (Seldaek) [inline src/Composer/Package/CompletePackageInterface.php:52]: This too could be simply string[] on the `@return`, no need for psalm annotations

**Comment by clxmstaab** [comment]
adjusted per Jordi's feedback

**Comment by Seldaek** [comment]
Thanks

**Comment by Seldaek** [comment]
As for the baseline yeah you can give it a go if it doesn't end up being a 2MB file maybe it's worth it :) I kinda imagine it'll be worse here than on composer/semver tho.

**Pull request merged**